fluent-syntax/src/stream.js

```diff
@@ -24,6 +26,7 @@ export class ParserStream {
     }
 
     this.index++;
+    this.checkLineBreak();
```

Comment:
I think it breaks the separation of concerns. In my design, Streams operate on... streams of characters. They don't keep the state of anything except the pointer in the Iterator and the buffer.

With your change, the stream becomes much more "aware". It now tracks state of the file. I feel like I'd prefer that not to happen in the stream for reasons mentioned above.

An example of where it would not work well is Rust where we can use CharStream to skip to a given character without using next. That would make the stream code per-implementation while so far I w as able to keep it parallel across implementations.

So, it's a bit hard, but I *think* that yes, I'd suggest taking line no for each entry separately, or, keep the pos in the AST and calculate line/col only when displaying.